src/kmedoids_ucb.cpp

```diff
@@ -107,24 +107,22 @@ int KMedoids::getSteps() {
  *  @param loss Loss function to be used e.g. L2
  */
 void KMedoids::setLossFn(std::string loss) {
-  if (loss.at(0) == 'L') {
-      loss = loss.substr(1);
-  }
-  if (loss == "manhattan") {
-      lossFn = &KMedoids::manhattan;
-  } else if (loss == "cos") {
-      lossFn = &KMedoids::cos;
-  } else if (loss == "inf") {
-      lossFn = &KMedoids::LINF;
-  } else if (std::isdigit(loss.at(0))) {
-      lossFn = &KMedoids::LP;
-      lp     = atoi(loss.c_str());
-      //std::stringstream st(loss);
-      //st >> lp;
-      //lp     = std::stoi(loss);
-  } else {
-      throw "unrecognized loss function";
-  }
+  try {
```

Comment:
Sorry to backtrack on this, but I still think we need the parsing of the loss in this function, for the same reason we include the other `if` statements below:

This is because we should not assume that the loss parsing is done by the `main` function in `main.cpp`. In fact, when called by Python, we do not call `main` -- we actually create a binding that calls `setLossFn` directly. For that reason, we should include the loss-parsing code *here*, not in `main.cpp`.

Furthermore, to avoid duplicate code, we should have `main` call *this* function to set the loss function. Right now, [this](https://github.com/ThrunGroup/BanditPAM/blob/03f93356830ee5ed2dfc4a905edf349619faa76e/src/main.cpp#L60-L63) only works for numeric loss functions and doesn't support `manhattan` etc. This will also be fixed once we have `main` call `setLossFn`

Reply:
Thanks @motiwari for this important point on Python! I have added back the loss parsing in `setLossFn` and retested all the cases in both C++ and Python:
```
1. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L2: Medoids: 694,800,306,714,324,959,737,527,168,251
2. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L15: Medoids: 194,319,558,60,133,896,448,475,251,354
3. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l manhattan: Medoids: 737,246,527,168,714,306,838,324,959,800
4. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l cos: Medoids: 567,182,839,340,385,979,915,758,609,649
5. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L15p: error: unrecognized loss function
```
Both of them give expected results for all cases. For duplication of code, maybe we could open a new issue and fix it there?

Reply:
Sounds good! @mailology could you file another issue for the duplicate code?